src/components/Common/Sidebar.js

```diff
@@ -1,346 +1,327 @@
-import React from 'react'
-import { Link, useNavigate } from 'react-router-dom'
-import menu from '../Data/menu.json'
-import menu2 from '../Data/menu2.json'
-import { _base, menuUrl, userSessionData } from '../../settings/constants'
-import { getMenu } from '../../services/OtherService/MenuService'
-import axios from 'axios'
-import UserService, { getUser } from '../../services/MastersService/UserService'
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useNavigate } from 'react-router-dom';
 
-class Sidebar extends React.Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      isSidebarMini: false,
-      isOpenMenu2: false,
-      menuData: [],
-      darkLightMode: 'light',
-      updateRtl: false,
-      redirect: false,
-      employeeMaster: [],
-      user_id: localStorage.getItem('id'),
-      role_id: localStorage.getItem('role_id'),
-      account_for: ''
-    }
-  }
+import menu from '../Data/menu.json';
+import menu2 from '../Data/menu2.json';
+import { _base } from '../../settings/constants';
+
+import {
+  getEmployeeListThunk,
+  getMenuListThunk
+} from '../../redux/services/Sidebar';
+
+const Sidebar = ({ activekey }) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [isSidebarMini, setIsSidebarMini] = useState(false);
 
- 
+  const [darkLightMode, setDarkLightMode] = useState('light');
+  const [updateRtl, setUpdateRtl] = useState(false);
+  const [user_id, setUser_id] = useState(localStorage.getItem('id'));
+  const [role_id, setRole_id] = useState(localStorage.getItem('role_id'));
+  const [account_for, setAccount_for] = useState('');
+  const { menuList, filterMenuList } = useSelector((state) => state?.sidebar);
```

Comment:
state sequency should be in proper way and add comment also

Reply:
resolved